app.py

```diff
@@ -849,17 +896,55 @@ def update_street_name(*args):
         *selected_streets, orientation = args
         street = selected_streets[list(SELECTED_STREET_DIVS.keys()).index(orientation)]
         try:
+            
             from_to = BASELINE[(BASELINE['street'] == street[0]) &
                                (BASELINE['direction'] == DIRECTIONS[orientation][dir_id])][['from_intersection',
                                                                                'to_intersection']].iloc[0]
+            for n, i in enumerate(from_to):
+                if i in ('Yonge', 'Bathurst', 'Front', 'Dundas'):
+                    from_to[n] = from_to[n] + ' St.'
+                elif i == 'Blue Jays':
+                    from_to[n] = 'Blue Jays Way'
+            
         except IndexError:
             return html.Div(className = 'nodata')
         else:
-            return [html.B(street[0] + ' ' + DIRECTIONS[orientation][dir_id] + ': '),
+            return [html.B(DIRECTIONS[orientation][dir_id] + ': '),
                     'from ' + from_to['from_intersection'] + ' to ' + from_to['to_intersection']]
 
 [create_update_street_name(i) for i in [0,1]]
 
+@app.callback(Output('streetss', 'children'),
+                  [*[Input(div_id, 'children') for div_id in SELECTED_STREET_DIVS.values()],
+                   Input('tabs', 'value'),
+                   Input(CONTROLS['timeperiods'], 'value'),
+               Input(CONTROLS['day_types'], 'value')])
+def update_street_name(*args):
+        #Use the input for the selected street from the orientation of the current tab
+    *selected_streets, orientation, timeperiod, day_type = args
+    street = selected_streets[list(SELECTED_STREET_DIVS.keys()).index(orientation)]
+    if street[0] in ('Front', 'Richmond', 'Queen', 'Wellington', 'Dundas', 'Adelaide', 'Bathurst'):
+        main_name = street[0] + ' Street'
+    elif street[0] in ('Spadina', 'University'):
+        main_name = street[0] + ' Avenue' 
+    time_range = TIMEPERIODS[(TIMEPERIODS['period'] == timeperiod) & (TIMEPERIODS['day_type'] == day_type)].iloc[0]['period_range']
+    if time_range == '(07:00:00-10:00:00)':
```

Comment:
Can you do this in PostgreSQL instead? It makes the Python code more resilient to changing data